TSPL.docc/LanguageGuide/Concurrency.md

```diff
@@ -1591,6 +1591,370 @@ You can also use an unavailable conformance
+    static func ==(lhs: Person, rhs: Person) -> Bool {
+        return lhs.id == rhs.id
+    }
+}
+```
+
+This allows the implementation of the conformance
+to use global actor isolated state
+while ensuring that state is only accessed
+from within the actor.
+
+### Inferring an Isolated Conformance
+
+> Note:
+> Isolated conformance inference is an upcoming language feature.
+> To enable it in current language modes of Swift,
+> use the feature identifier `InferIsolatedConformances`.
+
+Isolated conformances are inferred
+for global actor isolated types.
+The following code example declares a conformance to `Equatable`
+for a main-actor isolated class,
+and Swift infers main-actor isolation for the conformance:
+
+```swift
+@MainActor
+class Person {
+    var id: Int
+}
+
+// Inferred to be a @MainActor conformance to Equatable
+extension Person: Equatable {
+    static func ==(lhs: Person, rhs: Person) -> Bool {
+        return lhs.id == rhs.id
+    }
+}
+```
+
+You can opt out of this inference for a global-actor-isolated type
+by explicitly declaring that a protocol conformance is nonisolated.
+The following code example declares
+a nonisolated conformance to `Equatable` in an extension:
+
+```swift
+@MainActor
+class Person {
+    let id: Int
+}
+
+extension Person: nonisolated Equatable {
+    nonisolated static func ==(lhs: Person, rhs: Person) -> Bool {
+        return lhs.id == rhs.id
+    }
+}
+```
+
+### Data-Race Safety for Isolated Conformances
+
+Swift prevents data races for isolated conformances
+by ensuring that protocol requirements are only used
+on the global actor that the conformance is isolated to.
+In generic code,
+where the concrete conforming type is abstracted away,
+protocol requirements can be used through type parameters or `any` types.
+
+#### Using Isolated Conformances
+
+A conformance requirement to `Sendable`
+allows generic code to send parameter values to concurrently-executing code.
+If generic code accepts non-`Sendable` types,
+then the generic code can only use the input values
+from the current isolation domain.
+These generic APIs can safely accept isolated conformances
+and call protocol requirements
+as long as the caller is on the same global actor
+that the conformance is isolated to.
+The following code has a protocol `Dancer`,
+a class `Ballerina` with a main-actor isolated conformance to `Dancer`,
+and calls to the `Dancer.perform` requirement
+from a main-actor task and a concurrent task:
+
+```swift
+protocol Dancer {
+    func perform()
+}
+
+func startRecital(_ dancer: some Dancer) {
+    dancer.perform()
+}
+
+@MainActor class Ballerina: @MainActor Dancer { ... }
+
+Task { @MainActor in
+    let ballerina = Ballerina()
+    startRecital(ballerina)
+
+    let dancer: any Dancer = ballerina
+    dancer.perform()
+}
+
+Task { @concurrent in
+    let ballerina = Ballerina()
+    startRecital(c)  // Error
+
+    let dancer: any Dancer = ballerina // Error
```

Comment:
If there's a way to briefly state the error, let's include that in the comment.  Not all `// Error` comments in TSPL do this, but it's preferred when we can.